src/main/java/com/fasterxml/jackson/databind/type/TypeBindings.java

```diff
@@ -450,7 +457,7 @@ final static class AsKey {
         public AsKey(Class<?> raw, JavaType[] params, int hash) {
             _raw = raw ;
             _params = params;
-            _hash = hash;
+            _hash = 31 * Objects.hashCode(raw) + hash;
```

Comment:
Why not `raw.hashCode()`? Can it be null?

Reply:
I suspect that `raw.hashCode()` is sufficient, but I opted for Objects.hashCode just in case something I'm not aware of calls `asKey` with a null value.